crates/bin/c8y_configuration_poc/src/http_ext.rs

```diff
@@ -0,0 +1,72 @@
+use async_trait::async_trait;
+use tedge_actors::{
+    new_mailbox, Actor, Address, ChannelError, Mailbox, Message, Recipient, RuntimeError,
+    RuntimeHandle, Sender,
+};
+
+#[derive(Clone, Debug, Default, Eq, PartialEq)]
+pub struct HttpConfig {}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct HttpRequest {}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct HttpResponse {}
+
+/// Create a new HTTP connection managed behind the scene by an actor
+///
+/// This connection is private,
+/// i.e only the callee of `new_private_connection()` will be able to interact with it.
+///
+/// ```
+///       client                    http_con
+///             --------------------->|||| ============> http://host
+///         ||||<---------------------
+/// ```
+pub async fn new_private_connection(
+    runtime: &mut RuntimeHandle,
+    config: HttpConfig,
+    client: Recipient<HttpResponse>,
+) -> Result<Recipient<HttpRequest>, RuntimeError> {
+    let (mailbox, address) = new_mailbox(10);
+
+    let actor = PrivateHttpActor::new(config);
+    runtime.run(actor, mailbox, client).await?;
+
+    Ok(address.as_recipient())
+}
+
+struct PrivateHttpActor {
```

Comment:
Any specific reason behind the prefix `Private`?

Reply:
Just to stress that the connection is not shared.